### PR TITLE
[SEMVER-MAJOR] Remove legacyExplorer

### PR DESCRIPTION
**Pull request by gunjpan**
- Removes backward compatibility
for legacy end points `/models` & `/routes`
- Removes `legacyExplorer` flag which
enabled these routes
- Update related tests & tests using the
legacyExplorer flag

Connect to https://github.com/strongloop/loopback/issues/2037

### CODE REVIEW

**Comment by gunjpan** [comment]
@bajtos : PTAL, thanks.

File: test/relations.integration.js [hunk 1]
```diff
@@ -1652,7 +1652,7 @@ describe('relations - integration', function() {
           expect(res.body).to.be.an('object');
           expect(res.body.error).to.be.an('object');
           expect(res.body.error.name).to.equal('Error');
-          expect(res.body.error.statusCode).to.equal(500);
+          expect(res.statusCode).to.equal(500);
```
Comment (bajtos) [inline test/relations.integration.js]: This change looks unrelated to me, why is it needed?

File: test/rest.middleware.test.js [hunk 2]
```diff
@@ -197,42 +196,15 @@ describe('loopback.rest', function() {
...
-
-        expect(res.body).to.eql({});
-
-        done();
-      });
-  });
 
-  it('should report 404 for disabled legacy explorer route /routes', function(done) {
-    app.set('legacyExplorer', false);
+  it('should report 404 for disabled legacy route /routes', function(done) {
```
Comment (bajtos) [inline test/rest.middleware.test.js]: I think this test can be removed too.

File: test/rest.middleware.test.js [hunk 3]
```diff
@@ -197,42 +196,15 @@ describe('loopback.rest', function() {
...
     app.use(loopback.rest());
     request(app).get('/routes')
       .expect(404)
       .end(done);
   });
 
-  it('should report 404 for disabled legacy explorer route /models', function(done) {
-    app.set('legacyExplorer', false);
+  it('should report 404 for disabled legacy route /models', function(done) {
```
Comment (bajtos) [inline test/rest.middleware.test.js]: ditto

**Comment by bajtos** [comment]
Looks mostly good, left few comments to address.

File: test/relations.integration.js (hunk 1 above)
Comment (gunjpan) [inline test/relations.integration.js]: @bajtos : it was causing the test cases to fail on local. and I guess, it's the same reported by @loay in his [PR#2411](https://github.com/strongloop/loopback/pull/2411/commits/32318ec990f05ab638de811717c3e8637a57c8fa#diff-295cd28b918f0a0d539d8fd920b5c1f2R1655).

I will remove it from the context of this commit.

**Comment by gunjpan** [comment]
@bajtos : PTAL one more time, feedback applied.

**Comment by bajtos** [comment]
> it was causing the test cases to fail on local. 

Could you please work with @davidcheung to identify the cause of those failures and fix them.

**Comment by bajtos** [comment]
The Travis CI is failing, is it because of the `statusCode` check we were discussing in https://github.com/strongloop/loopback/pull/2419#discussion_r66605859?

@gunjpan @davidcheung PTAL

**Comment by bajtos** [comment]
The code change LGTM, I just want to make sure the CI is passing before we land this change.

**Comment by davidcheung** [comment]
@slnode test please

**Comment by gunjpan** [comment]
@slnode test please

**Comment by gunjpan** [comment]
@slnode test please
